src/bidiMapper/domains/context/BrowsingContextImpl.ts

```diff
@@ -628,7 +628,8 @@ export class BrowsingContextImpl {
 
     return {
       navigation: cdpNavigateResult.loaderId ?? null,
-      url,
+      // Url can change due to redirect get the latest one.
+      url: wait === BrowsingContext.ReadinessState.None ? url : this.#url,
```

Comment:
We may need to await for something like `Frame.startedLoading` event and always resolve the url from `this` .
Will take a look in the spec of how that should behave.

Reply:
Waiting for the `Frame.startedLoading` would have minimal effect as the event does not contain enough information.